File: src/python/0301_0600/0453_minimum_moves_to_equal_array_elements.py
from typing import List


def minMoves1(nums: List[int]) -> int:
    res, min_val = 0, min(nums)

    for num in nums:
        res += num - min_val

    return res


def minMoves2(nums: List[int]) -> int:
    return sum(nums) - len(nums) * min(nums)


if __name__ == "__main__":
    nums = [1, 2, 3]
    print(minMoves1(nums))  # 3
    print(minMoves2(nums))  # 3

File: src/cpp/0301_0600/0453_minimum_moves_to_equal_array_elements.cc
#include <algorithm>
#include <iostream>
#include <vector>
using namespace std;

int minMoves(vector<int>& nums) {
    int res = 0;
    int min_num = *min_element(nums.begin(), nums.end());

    for (int num : nums) {
        res += num - min_num;
    }

    return res;
}

int main() {
    vector<int> nums = {1, 2, 3};
    cout << minMoves(nums) << endl;  // 3
    return 0;
}
